### PR TITLE
[tune] Fix Docs

### PR DESCRIPTION
**Pull request by richardliaw**
## What do these changes do?
 - Introduces Python2 and Python3 compatibility for the Tune Client.
 - removes hard dependencies on TensorFlow, Requests, so that users can use `Tune` by simply installing Ray.
 - Adds `requests` to testing dependency (although seems like this hasn't previously been an issue?)
 - Add parallel coordinate visualization to Tune Docs
## Related issue number
#1328, #1458

### CODE REVIEW

File: doc/source/tune.rst [hunk 1]
```diff
@@ -67,11 +67,16 @@ Visualizing Results
 
 Ray Tune logs trial results to a unique directory per experiment, e.g. ``~/ray_results/my_experiment`` in the above example. The log records are compatible with a number of visualization tools:
 
-To visualize learning in tensorboard, run:
+To visualize learning in tensorboard, install TensorFlow:
```
Comment (richardliaw) [inline doc/source/tune.rst]: tensorflow via pip automatically installs tensorboard

**Review by ericl — APPROVED** [review]

**Comment by richardliaw** [comment]
test this please

**Comment by richardliaw** [comment]
Jenkins and Linux Travis passed - Mac builds still queued; merging now.

**Comment by robertnishihara** [comment]
You can run Mac jobs faster by pushing to ray-private-travis.

**Pull request merged**